x/trigger/keeper/event_listener_test.go

```diff
@@ -0,0 +1,158 @@
+package keeper_test
+
+import (
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdktx "github.com/cosmos/cosmos-sdk/types/tx"
+	"github.com/provenance-io/provenance/x/trigger/types"
+)
+
+func (s *KeeperTestSuite) TestGetAndSetEventListener() {
+	var event types.TriggerEventI = &types.BlockHeightEvent{BlockHeight: 1}
+	actions, _ := sdktx.SetMsgs([]sdk.Msg{&types.MsgDestroyTriggerRequest{Id: 1, Authority: s.accountAddresses[0].String()}})
+	eventAny, _ := codectypes.NewAnyWithValue(event)
+	newTrigger := types.NewTrigger(1, s.accountAddresses[0].String(), eventAny, actions)
+	s.app.TriggerKeeper.SetEventListener(s.ctx, newTrigger)
+
+	tests := []struct {
+		name     string
+		id       types.TriggerID
+		expected *types.Trigger
+		prefix   string
+		err      string
+	}{
+		{
+			name:     "valid - event listener",
+			id:       1,
+			expected: &newTrigger,
+			prefix:   types.BlockHeightPrefix,
+			err:      "",
+		},
+		{
+			name:     "invalid - event listener doesn't exist",
+			id:       999,
+			expected: nil,
+			prefix:   types.BlockHeightPrefix,
+			err:      types.ErrEventNotFound.Error(),
+		},
+		{
+			name:     "invalid - event listener doesn't exist with wrong prefix",
+			id:       1,
+			expected: nil,
+			prefix:   types.BlockTimePrefix,
+			err:      types.ErrEventNotFound.Error(),
+		},
+	}
+
+	for _, tc := range tests {
+		s.Run(tc.name, func() {
+			listener, err := s.app.TriggerKeeper.GetEventListener(s.ctx, tc.prefix, tc.id)
+			if len(tc.err) > 0 {
+				s.Error(err)
+				s.ErrorContains(err, tc.err)
```

Comment:
The `ErrorContains` check does the `Error` check before checking the contents.

Also, except for very very very simple and obvious cases, it's best to always include a message in your assertions.

```suggestion
				s.ErrorContains(err, tc.err, "GetEventListener(%q, %d) error", tc.prefix, tc.id)
```

The messages help answer the question, "What am I looking at?" that usually comes to mind when you're looking at the results of a run of tests (e.g. in Github Actions). They also often make it easier to identify which assertion is failing; this comes into play when there are multiple similar assertions back-to-back.

Since there is usually many error checks involved in unit tests, it's best to always put the function call that generated the error as the message. Even better is providing some args, but only if the args are fairly short. Like, if it's a huge object, probably not worth including in the message (e.g. don't put the content of the context in there). But in this case, the prefix and id might be nice additions.

This comment about the `msgAndArgs` applies to all unit tests. Just get into the habit of always including them. They make life significantly easier the next time changes are needed or something breaks.